src/compression.rs

```diff
@@ -75,7 +75,7 @@ fn get_first_five<'a>(
     }
 }
 
-pub(crate) fn read_compression<'a>(
+pub fn read_compression<'a>(
```

Comment:
a cheeky suggestion is calling this `sniff` instead. You called it `detect_format` in the benchmarks, which I think it is also better than `read_compression`.

Reply:
But also... this ends up being really similar to `get_reader` (but with return values order switched), do you have other use cases to make it public?

Reply:
I like the name `sniff`.

In fact `sniff` and `get_reader`are different:
- `sniff` return compression fomat and the raw stream
- `get_reader` return the uncompress stream and the compression format.

So isn't the same function but ok I can change the order of return in `sniff` to have a cleaner interface.